Source/WebCore/html/canvas/WebGL2RenderingContext.cpp

```diff
@@ -591,11 +605,12 @@ void WebGL2RenderingContext::copyBufferSubData(GCGLenum readTarget, GCGLenum wri
         return;
 
     RefPtr<WebGLBuffer> readBuffer = validateBufferDataParameters("copyBufferSubData", readTarget, GraphicsContextGL::STATIC_DRAW);
+    if (!readBuffer)
+        return;
+
     RefPtr<WebGLBuffer> writeBuffer = validateBufferDataParameters("copyBufferSubData", writeTarget, GraphicsContextGL::STATIC_DRAW);
-    if (!readBuffer || !writeBuffer) {
-        synthesizeGLError(GraphicsContextGL::INVALID_OPERATION, "copyBufferSubData", "Invalid readTarget or writeTarget");
+    if (!writeBuffer)
```

Comment:
Is the "Invalid writeTarget" synthetic error not required here?

Reply:
`validateBufferDataParameters` or its callees generate a synthetic error before returning `nullptr`.